includes/widgets/class-wc-widget-cart.php

```diff
@@ -69,4 +73,39 @@ public function widget( $args, $instance ) {
 
 		$this->widget_end( $args );
 	}
+
+	/**
+	 * This method provides the JS script which will execute on addition of a new widget.
+	 *
+	 * @return void
+	 */
+	private function enqueue_ajax_script() {
+		$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+		wp_register_script( 'wc-cart-fragments', WC()->plugin_url() . '/assets/js/frontend/cart-fragments' . $suffix . '.js', array( 'jquery', 'js-cookie' ), WC_VERSION, true );
+
+		wp_localize_script( 'wc-cart-fragments', 'wc_cart_fragments_params',
+			array(
+				'ajax_url'      => WC()->ajax_url(),
+				'wc_ajax_url'   => WC_AJAX::get_endpoint( '%%endpoint%%' ),
+				'cart_hash_key' => apply_filters( 'woocommerce_cart_hash_key', 'wc_cart_hash_' . md5( get_current_blog_id() . '_' . get_site_url( get_current_blog_id(), '/' ) . get_template() ) ),
+				'fragment_name' => apply_filters( 'woocommerce_cart_fragment_name', 'wc_fragments_' . md5( get_current_blog_id() . '_' . get_site_url( get_current_blog_id(), '/' ) . get_template() ) ),
+			)
+		);
+
+		wp_enqueue_script( 'wc-cart-fragments' );
+
+		?>
+		<script type="text/javascript">
```

Comment:
Instead of outputting this directly can you use wc_enqueue_js function instead?

Reply:
Yea, I tried.... But the `wp_enqueue_script` isn't working the way it should work in customizer. That's why I reverted the changes. But anyway, I'll give it a try again.

Reply:
No I mean `wc_enqueue_js` it is a WooCommerce function for including inline scripts in code. So that whole <script part you added, what's inside you can just pass into `wc_enqueue_js` and it will take care of outputting it in the html code for you.

Reply:
@kloon Unfortunately `wc_enqueue_js` isn't working cause this gets called inside `wc_print_js` and this `wc_print_js` is hooked to `wp_footer` hook, which can't make external plugin's JS based AJAX working inside WP Customizer for lexical scope behavior reason of JS I think. Anyway, after all of this I tested with multiple hooks also, like-
* `customize_preview_init`
* `customize_controls_print_scripts`
* `customize_controls_enqueue_scripts`

**NO LUCK with those above hooks either**.

Lastly, I know this seems kinda hacky solution, but as the structure of the widget is different than other widgets(It uses a native WooCommerce file AJAX functionality to populate it's data) and this structure isn't compatible with WordPress and WooCommerce current JS enqueuing system for it's AJAX call in customizer preview, I would prefer to do it for right now with this hacky way. And in future if we find a better solution we'll do some more pull request. :slightly_smiling_face: 

> N.B. : We can add a comment with `\\ TODO:` to mark it to make it better in future.

Reply:
The `@todo` comment has been proposed in some later commits. It has been added to the *PHPDoc* of the method.

Reply:
@kloon Just a reminder. :eyes: This PR is been hanging for a long time, so would be great if you make a little time to review it. :smiley: